solarforecastarbiter/datamodel.py

```diff
@@ -1052,6 +1055,40 @@ class ProcessedForecastObservation(BaseModel):
     observation_values: Union[pd.Series, str, None]
 
 
+@dataclass(frozen=True)
+class MetricValue(BaseModel):
+    category: str
+    metric: str
+    index: str
+    value: float
```

Comment:
normalization value? or does that go in MetricResult?

Reply:
I can't think of anything to add here. I agree normalization should go here, and I think any value that comes from outside of the metrics package should as well.  @dplarson Is there any other inputs to any of the other metrics that is "user-define". 

One example from reviewing the metrics documentation... the ramping threshold in the example on the metrics page would be input from outside of the metrics, but probably the ramping event timeseries observations effectively would be calculated separately and not available to the metrics at all but still maybe that would be useful here?

Reply:
The only other "user-defined" input is financial cost in the valuation metrics (i.e. `metrics.valuation.fixed_error_cost(obs, fx, cost)`).

Reply:
Oh and, cost metrics might need their own data model since there will be significant amount of user input potentially, like timeseries of cost or at least a static $/MW value and/or thresholds for misforecasts etc.

Reply:
We could certainly add a cost parameter that could later encompass those more complicated costs. But, where do these parameters (cost, normalization, reference_forecast_id) really belong? If they can change with each metric and each category, then I would agree that they belong on MetricValue. If they a set per observation/forecast pair, then I think they should be on MetricResult. And if they can only be defined and applied to the report as a whole, they should probably be associated with RawReport (or some new metric dataclass that has the tuple of MetricResults).